WordPress/gutenberg
# Multiple use block validation logic improvement

## What?
Fixes: https://github.com/WordPress/gutenberg/issues/38502

## Why? 
Any block that doesn't support the `multiple` feature, doesn't display the **Multiple blocks** validation warning when it's inside group blocks.

## How?
Include `innerBlocks` in the validation process.

## Testing Instructions

1. Open post or page to edit
2. Insert "Group" block
3. Insert the "More" block inside the "Group" block
4. Duplicate "Group" block.

The "Read more" should display the validation warning.

## Screenshots or screencast 

Before:

https://user-images.githubusercontent.com/17360543/160422035-1111351a-8e70-45f4-b9c9-571cc0fd3bed.mov

After:

https://user-images.githubusercontent.com/17360543/160421847-bcc4d749-3e38-43e4-955e-35e258fe6534.mov
